### DISCUSSION TITLE
Git Ignore auto-generated

### DISCUSSION THREAD

**Discussion by Ohior** [pypa/virtualenv#2400] [Ideas]
virtualenv automatically generates a gitignore file. And this file gets generated inside of the environment folders. 
This gitignore file is hidden from users. It will be better if this file is generated outside of the environment folder and populated with python packages to be ignored.

**Comment by gaborbernat** [comment]
I disagree. I think this ignore file should be scoped for the virtual environment folder. This stops the user deleting it by accident.

**Reply by Ohior** [reply]
when I started using virtualenv, I usually create my own gitignore, not knowing that it was already created for me.

**Reply by gaborbernat** [reply]
gitignore files are not singular per repo. You can have multiple per file and only applies to that subfolder. This is expected and correct behaviour. You absolutely should not be reusing/changing the one we generate. That's handled and managed by virtualenv.

**Reply by Ohior** [reply]
Oh! Okay, I understand why now.

**Comment by gaborbernat** [comment]
This proposal was considered and a design decision was made. The `.gitignore` file is intentionally scoped to the virtual environment folder to prevent users from accidentally deleting it. This behavior is by design. Closing as answered.
